### DISCUSSION TITLE
Solution if private server on NAS inaccessible

### DISCUSSION THREAD

**Discussion by jmgroud** [rustdesk/rustdesk#9589] [Q&A]
### English
**---------------------------------------**

Hello

**Configuration**

I have just installed my private server on a Synology NAS with DSM 7.2.2.
Everything works perfectly with ‘Container manager’ on my NAS.
I have set up all the client machines with the value of the key given by this software and with the DNS name of the NAS.

**Question**

If my NAS shuts down for any number of reasons (breakdown, maintenance, etc.) or if external access via the Internet becomes impossible (problem with my provider, etc.), all my computers become inaccessible.
If I want to temporarily access a computer with the ‘RustDesk’ servers, I have to change the settings on 2 machines:
- The one I use to connect: This is possible here.
- The client machine on which I need to connect remotely: HERE IS A PROBLEM because I can't connect to this machine :-(

How can I access a machine previously set up with a connection via my NAS if the NAS is no longer accessible?
Thank you in advance for your solutions.

### Français
**---------------------------------------**

Bonjour

**Configuration**

Je viens d'installer mon serveur privé sur un NAS Synology avec DSM 7.2.2.
Tout fonctionne parfaitement avec "Container manager" sur mon NAS.
J'ai paramétré l'ensemble des machines clientes avec la valeur de la clé donné par ce logiciel et avec le nom DNS du NAS.

**Question**

Si mon NAS s'arrête pour de multiples raisons (panne, maintenance, ...) ou si son accès externe par Internet devient impossible (problème chez mon fournisseur, ...), tous mes ordinateurs deviennent inaccessibles.
Si je désire accéder temporairement à un ordinateur avec les serveurs de "RustDesk", je dois modifier le paramétrage de 2 machines :
- Celle que j'utilise pour me connecter :  Ici c'est possible.
- La machine cliente sur laquelle je dois me connecter à distance : ICI IL Y A UN PROBLEME car je ne peux pas me connecter sur cette machine :-(

Comment puis-je accéder à une machine précédemment paramétrée avec une connexion par mon NAS si ce NAS n'est plus accessible ?
Merci d'avance pour vos solutions.

**Comment by ian-barber** [comment]
If you untick the Always connect via relay option, would it default to the public server ?

![Screenshot 2024-10-08 at 15 45 52](https://github.com/user-attachments/assets/8b4743b3-0e65-460e-a45e-c89dd45dfdc9)

**Reply by jmgroud** [reply]
Hello ian-barber

### English
-----------------------------------
Good evening
This solution doesn't seem to work...
I stopped the ‘rustdesk-server’ service on the NAS to simulate a failure, then unchecked the ‘Force relay connection’ option on my laptop.
When I test the connection to a client, I get the following error:

Connection error
Failed to connect to <address>:<port>: Please try again later

### Français
-----------------------------------
Bonsoir
Cette solution ne semble pas fonctionner...
J'ai arrêté le service 'rustdesk-server' sur le NAS pour simuler une panne puis ai décoché sur mon portable l'option 'Forcer la connexion relai'.
En testant la connexion vers un client, j'ai l'erreur suivante :

Erreur de connexion
Failed to connect to <adresse>:<port>: Veuillez essayer plus tard

**Comment by ian-barber** [comment]
I am wondering what this checkbox option is therefore, maybe someone from RD could throw some light on it

**Reply by jmgroud** [reply]
Yes, me too ;-)

**Comment by basilgello** [comment]
Let me try to clarify the problem: you want to access computers inside your intranet from inside and outside of it, correct?

If there are not so many computers, why not deploy [Yggdrasil](https://github.com/yggdrasil-network/yggdrasil-go) (decentralized overlay IPv6 network) and bind the listening port 21114 of each Rustdesk computer solely to the IPv6 TUN interface provided by Yggdrasil? Then you connect via Ygg IPv6 address and port instead of Rustdesk ID and you can ditch Rustdesk Server setup completely.

This way you get self-healing, always-encrypted VPN on top of your Intranet AND you can listen the Yggdrasil node (aka "peer") to the Internet and protect it with password to organize easy outside access like a true VPN.

**Comment by jmgroud** [comment]
hello basilgello
Too complicated for me :-(

**Comment by xlionjuan** [comment]
If High Availability is you desired, and you can't keep your Synology NAS High Availability, you should consider to rent a VPS, cheapest one is enough for most use cases.

* Rent a VPS, choose Ubuntu 24.04 as your system
* Deploy RustDesk Server though Docker(You can migrate your current one)
* Use Watchtower to update your RustDesk Server container automatically
* Enable unattended-upgrades on your Ubuntu system and enable automatic reboot

**Reply by jmgroud** [reply]
### English
**---------------------------------------**
Tanks a lot xlionjuan.
This is indeed a solution, but if possible I'd like a solution that would allow me to keep my NAS and, in the event of unavailability, have a process for continuing to connect to my customers.

### Français
**---------------------------------------**
C'est en effet une solution mais si possible, j'aimerai une solution qui pourrait me faire garder mon NAS et dans le cas d'une indisponibilité, avoir un processus pour pouvoir continuer à me connecter svers mes clients.

**Reply by xlionjuan** [reply]
For me, RustDesk for main, Anydesk or TeamViewer for secondary backup.

**Comment by ian-barber** [comment]
I also run the server on my Synology and I can understand where you are coming from but even if there was a checkbox which switched to a public server, you would still need each client to remove the NAS information they already have in the Network settings of Rust Desk.

<img width="523" alt="Screenshot 2024-10-12 at 19 23 12" src="https://github.com/user-attachments/assets/2f4cd5a1-7a53-4a74-ba5c-682e877d5ed5">
